app/services/ping_thing_user_stats_service.rb

```diff
@@ -0,0 +1,65 @@
+class PingThingUserStatsService
+
+  def initialize(network: "mainnet", interval: 5)
+    @network = network
+    @interval = interval
+    @ping_things = []
+    @stats = []
+  end
+
+  def call
+    gather_ping_things
+    delete_old_stats
+    pt_by_user.each do |user_id, user_ping_things|
+      ping_times = user_ping_things.pluck(:response_time).compact.sort
+      username = User.find(user_id).username
+
+      pts = PingThingUserStat.find_or_create_by(
+        user_id: user_id,
+        interval: @interval,
+        network: @network
+      )
+      pts.update(
+        username: username,
+        median: ping_times.median,
+        min: ping_times.min,
+        max: ping_times.max,
+        p90: ping_times.first((ping_times.count * 0.9).to_i).last,
+        num_of_records: ping_times.count,
+        average_slot_latency: user_ping_things.map{ |pt| pt.slot_landed && pt.slot_sent ? pt.slot_landed - pt.slot_sent : nil }.compact.average&.round(1),
```

Comment:
```suggestion
        average_slot_latency: user_ping_things.map do |pt| 
          pt.slot_landed && pt.slot_sent ? pt.slot_landed - pt.slot_sent : nil
        end.compact.average&.round(1),
```

Reply:
Wrong indentation (missing one space in line 32 and 33)